inc/functions_search.php

```diff
@@ -1294,7 +1363,7 @@ function perform_search_mysql($search)
 			$query = $db->query("
 				SELECT t.tid, t.firstpost
 				FROM ".TABLE_PREFIX."threads t
-				WHERE 1=1 {$thread_datecut} {$thread_replycut} {$thread_prefixcut} {$forumin} {$thread_usersql} {$permsql} {$visiblesql} AND t.closed NOT LIKE 'moved|%' {$subject_lookin}
+				WHERE 1=1 {$thread_datecut} {$thread_replycut} {$thread_prefixcut} {$forumin} {$thread_usersql} {$permsql} {$visiblesql} AND {$unapproved_where_t} AND t.closed NOT LIKE 'moved|%' {$subject_lookin}
```

Comment:
Hmm, I think `{$unapproved_where_t}` should be parenthesized, because there might be one or more `OR ...` in it. Am I right?
```suggestion
				WHERE 1=1 {$thread_datecut} {$thread_replycut} {$thread_prefixcut} {$forumin} {$thread_usersql} {$permsql} {$visiblesql} AND ({$unapproved_where_t}) AND t.closed NOT LIKE 'moved|%' {$subject_lookin}
```

Reply:
The OR... one's get wrapped in parentheses so isn't necessary I don't think

Reply:
Actually I've changed my mind as it could prevent issues if the function is modified later.

Reply:
Really sorry for that, actually the initial PR/commit is fine with the SQL queries, no parentheses should be added.

I have another concern here. Will it work in specific forums if a user (not a moderator) is set to `Can only view own threads` in the forums? That means this user should be able to view threads with `visible=0` regardless of their moderating privileges.

And further, will it be better to merge `$unapproved_where_t` into `$permsql`, probably somewhere in here?
https://github.com/mybb/mybb/blob/9feef312dc22abef36dcc1ddb1a703e66d9c2fe1/inc/functions_search.php#L1234-L1248

Reply:
> I have another concern here. Will it work in specific forums if a user (not a moderator) is set to `Can only view own threads` in the forums? That means this user should be able to view threads with `visible=0` regardless of their moderating privileges.

`Can only view threads?` doesn't have any effect based on `visible` field, only if the user was the thread author.

> And further, will it be better to merge `$unapproved_where_t` into `$permsql`, probably somewhere in here?

Best to keep separate to follow how the code was structured already imo.

Reply:
Hmm, sorry I was thinking/speaking in the wrong way.

It's not related to `Can only view own threads` or `Can only view threads` but users should be able to see their own threads/posts even if the threads/posts are unapproved (`visible=0`).

Reply:
Normal users aren't allowed to view unapproved posts even if they are the poster. I've just tried on a test board with several different tests eg moderate new posts: for a user, for a usergroup, in a forum. User is told their posts require moderation but they cannot view the unapproved posts.

Reply:
Hmm. Looks like we may both have some misunderstanding about it.. There's an option **Access Own Unapproved Threads / Posts** : Allow general users to access their own unapproved threads and posts, which controls that behavior.

I thought that users are able to see their own unapproved threads/posts is baked in MyBB because I set it to yes 🤣 but the default setting of it is no actually.

But.. if you read something like `The post made by you is under moderation and currently not visible publicly. The post will be visible to everyone once a moderator approves it.`, the post is collapsed and the user can read the post content by click "Show this post" button on the right.

Reply:
I was not aware of this setting!

I've done some quick testing and if that setting is enabled since the thread is hidden everywhere except for the forum list it seems. User CP latest threads, thread subscriptions etc. Users also cannot interact with it whilst unapproved (subscribing to it for example).

If we want to press ahead with those threads/posts in search I think it is a simple enough change:
```php

	// Normal users
	if($mybb->user['uid'] > 0 && $mybb->settings['showownunapproved'] == 1)
	{
		return "({$aliasdot}visible = 1 AND {$aliasdot}uid = {$mybb->user['uid']})";
	}
	return "{$aliasdot}visible = 1";
```

Reply:
I've implemented in 48ae224

Noticed when clicking through to the posts that there are jGrowl notices saying the post doesn't exist. Maybe it is best not to include this change actually 🤔 Perhaps it is intended you only find your unapproved posts by looking for them in the forum/thread display??

Reply:
> Noticed when clicking through to the posts that there are jGrowl notices saying the post doesn't exist. Maybe it is best not to include this change actually 🤔 Perhaps it is intended you only find your unapproved posts by looking for them in the forum/thread display??

That seems a bug to me, what do you think? Another bug report?

And I have a related bug about showing post doesn't exist months ago but just can't remember if I had reported or not or what it was really about...